### PR TITLE
Add CSV weather source

### LINKED ISSUE #223: Add Weather Sources

### PR DESCRIPTION
**Pull request by mia-krause**
resolves #223 for CSV

I had a deeper look into fast csv processing with java 8 and came to the conclusion that we already have the best [solution](https://github.com/ie3-institute/PowerSystemDataModel/blob/d2ec004d530c5093240cc6ece36afd30cb836287/src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java#L400-L408) for two reasons:
1. You're not really supposed to read csv files in parallel, as this [answer](https://stackoverflow.com/questions/11098873/how-to-split-a-csv-file-into-multiple-chunks-and-read-those-chunks-in-parallel-i#comment14563911_11098988) and a few others illustrate. And we are not reading in parallel as we have a single reader.
2. After reading, you're free to process the rows in parallel, which is also what we're doing. `reader.lines()` is lazily populated, so our parallel parsing threads can "grab" a line once it is read in instead of waiting for the reader to finish.
I don't see any optimization here, except _maybe_ in form of some [framework](https://github.com/uniVocity/csv-parsers-comparison).

### CODE REVIEW

**Comment by johanneshiry** [comment]
@mia-krause pls let me know when it is ready for review :)

**Comment by mia-krause** [comment]
@johanneshiry should be ready

File: src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java [hunk 1]
```diff
@@ -243,6 +249,13 @@ public BufferedReader initReader(String filePath) throws FileNotFoundException {
       IndividualTimeSeriesMetaInformation individualMetaInformation =
           (IndividualTimeSeriesMetaInformation) metaInformation;
 
+      if (columnSchemes != null
+          && columnSchemes.length > 0
+          && Stream.of(columnSchemes)
+              .noneMatch(scheme -> scheme.equals(individualMetaInformation.getColumnScheme()))) {
+        return Optional.empty();
```
Comment (johanneshiry) [inline src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java:256]: You might consider making this a little bit more verbose by adding a logged warning or error?

**Review by johanneshiry — CHANGES_REQUESTED** [review]

**Review by johanneshiry — APPROVED** [review]

**Comment by johanneshiry** [comment]
Pls let @ckittl take a look at this as well before merging as most of the time series concepts are his work.

File: src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy [hunk 2]
```diff
@@ -44,66 +33,45 @@ class CsvTimeSeriesSourceIT extends Specification implements CsvTestDataMeta {
...
-		source = new CsvTimeSeriesSource(";", timeSeriesFolderPath, new FileNamingStrategy(), coordinateSource)
+		source = new CsvTimeSeriesSource(";", timeSeriesFolderPath, new FileNamingStrategy())
 	}
 
 	def "The csv time series source is able to provide an individual time series from given field to object function"() {
 		given:
-		def weatherValueFunction = { fieldToValues -> source.buildWeatherValue(fieldToValues) }
-		def tsUuid = UUID.fromString("8bc9120d-fb9b-4484-b4e3-0cdadf0feea9")
-		def filePath = new File(this.getClass().getResource( File.separator + "testTimeSeriesFiles" + File.separator + "its_weather_8bc9120d-fb9b-4484-b4e3-0cdadf0feea9.csv").toURI())
+		def weatherValueFunction = { fieldToValues -> source.buildTimeBasedValue(fieldToValues, HeatAndSValue, new TimeBasedSimpleValueFactory<>(HeatAndSValue)) }
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy:45]:
```suggestion
		def heatAndSValueFunction = { fieldToValues -> source.buildTimeBasedValue(fieldToValues, HeatAndSValue, new TimeBasedSimpleValueFactory<>(HeatAndSValue)) }
```

File: src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy [hunk 3]
```diff
@@ -44,66 +33,45 @@ class CsvTimeSeriesSourceIT extends Specification implements CsvTestDataMeta {
 	CsvTimeSeriesSource source
 
 	def setupSpec() {
-		defaultCoordinate = GeoUtils.DEFAULT_GEOMETRY_FACTORY.createPoint(new Coordinate(7.4116482, 51.4843281))
-		coordinateSource = Mock(IdCoordinateSource)
-		coordinateSource.getCoordinate(5) >> Optional.of(defaultCoordinate)
+		coordinateSource = WeatherTestData.coordinateSource
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy:36]: I think the shared coordinate source isn't even used anymore, isn't it?

**Review by ckittl — CHANGES_REQUESTED** [review]
Despite the two minor naming issues in the test, this code look good to me!

File: src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy (hunk 3 above)
Comment (johanneshiry) [reply src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSourceIT.groovy:36]: you're right - adapted it accordingly

**Review by ckittl — APPROVED** [review]

**Pull request merged**